smif/convert/interval.py

```diff
@@ -433,7 +409,6 @@ def register(self, intervals, set_name):
         self._register[set_name] = OrderedDict()
 
         for interval in intervals:
-
```

Comment:
We need to harmonise the API across `interval` and `region`.  `IntervalRegister.register` requires `intervals` and a `set_name` as arguments.  `RegionRegister.register` requires only `region_set`.  The `RegionSet` class contains a `name` property, where as `intervals` are of type `list`.

Two options:
* leave as is
* make another class `IntervalSet` which encapsulates a list of intervals, and an interval-set name string

The latter will also require removing the iteration over interval list in the `IntervalRegister.register` method.

Finally, the `IntervalRegister.convert` method should move closer to using coefficients as the `RegionRegister.convert` method does.